### PR TITLE
Fix regression due to refactor in ZigZagRandomTarget

### PR DESCRIPTION
**Pull request by fedpet**

### CODE REVIEW

**Comment by DanySK** [comment]
@fedpet having the regression fix is great, but it would be much better to also have a test that intercepts this future regression: can you provide one?

**Comment by fedpet** [comment]
> @fedpet having the regression fix is great, but it would be much better to also have a test that intercepts this future regression: can you provide one?

Done

File: alchemist/alchemist-smartcam/src/main/kotlin/it/unibo/alchemist/model/implementations/movestrategies/ZigZagRandomTarget.kt [hunk 1]
```diff
@@ -46,7 +46,10 @@ class ZigZagRandomTarget<T> (
         return computeTarget()
     }
 
-    private fun changeDirection() = env.makePosition(2 * rng.nextDouble() - 1, 2 * rng.nextDouble() - 1)
```
[the remark this answers is not in the document]
Comment (fedpet) [reply alchemist/alchemist-smartcam/src/main/kotlin/it/unibo/alchemist/model/implementations/movestrategies/ZigZagRandomTarget.kt]: One returns the direction while the other does not

[the remark this answers is not in the document]
Comment (fedpet) [reply alchemist/alchemist-smartcam/src/main/kotlin/it/unibo/alchemist/model/implementations/movestrategies/ZigZagRandomTarget.kt]: > One returns the direction while the other does not

Correction: one changes the directions while the other only returns it

[the remark this answers is not in the document]
Comment (DanySK) [reply alchemist/alchemist-smartcam/src/main/kotlin/it/unibo/alchemist/model/implementations/movestrategies/ZigZagRandomTarget.kt]: Could be rewritten more functionally by using `also`, but it's ok.

**Pull request merged**